src/main/java/org/breedinginsight/brapi/v2/services/BrAPITrialService.java

```diff
@@ -52,4 +93,278 @@ private long countGermplasm(UUID programId, String trialDbId) throws ApiExceptio
+        row.put(ExperimentObservation.Columns.ENV_LOCATION, Utilities.removeProgramKey(study.getLocationName(), program.getKey()));
+        BrAPISeason season = seasonDAO.getSeasonById(study.getSeasons().get(0), program.getId());
+        row.put(ExperimentObservation.Columns.ENV_YEAR, season.getYear());
+        row.put(ExperimentObservation.Columns.EXP_UNIT_ID, Utilities.removeProgramKeyAndUnknownAdditionalData(ou.getObservationUnitName(), program.getKey()));
+
+        // get replicate number
+        Optional<BrAPIObservationUnitLevelRelationship> repLevel = ou.getObservationUnitPosition()
+                .getObservationLevelRelationships().stream()
+                .filter(level -> BrAPIConstants.REPLICATE.getValue().equals(level.getLevelName()))
+                .findFirst();
+        repLevel.ifPresent(brAPIObservationUnitLevelRelationship ->
+                row.put(ExperimentObservation.Columns.REP_NUM, Integer.parseInt(brAPIObservationUnitLevelRelationship.getLevelCode())));
+
+        //get block number
+        Optional<BrAPIObservationUnitLevelRelationship> blockLevel = ou.getObservationUnitPosition()
+                .getObservationLevelRelationships().stream()
+                .filter(level -> BrAPIConstants.BLOCK.getValue().equals(level.getLevelName()))
+                .findFirst();
+        blockLevel.ifPresent(brAPIObservationUnitLevelRelationship ->
+                row.put(ExperimentObservation.Columns.BLOCK_NUM, Integer.parseInt(brAPIObservationUnitLevelRelationship.getLevelCode())));
+        if (ou.getObservationUnitPosition() != null && ou.getObservationUnitPosition().getPositionCoordinateX() != null &&
+                ou.getObservationUnitPosition().getPositionCoordinateY() != null) {
+            row.put(ExperimentObservation.Columns.ROW, Double.parseDouble(ou.getObservationUnitPosition().getPositionCoordinateX()));
+            row.put(ExperimentObservation.Columns.COLUMN, Double.parseDouble(ou.getObservationUnitPosition().getPositionCoordinateY()));
+        }
+        if (ou.getTreatments() != null && !ou.getTreatments().isEmpty()) {
+            row.put(ExperimentObservation.Columns.TREATMENT_FACTORS, ou.getTreatments().get(0).getFactor());
+        } else {
+            row.put(ExperimentObservation.Columns.TREATMENT_FACTORS, null);
+        }
+        row.put(ExperimentObservation.Columns.OBS_UNIT_ID, ouId);
+
+        return row;
+    }
+
+
+
+    private void addObsVarColumns(
+            List<Column> columns,
+            List<BrAPIObservationVariable> obsVars,
+            boolean includeTimestamps,
+            Program program) {
+        for (BrAPIObservationVariable var: obsVars) {
+            Column obsVarColumn = new Column();
+            obsVarColumn.setDataType(Column.ColumnDataType.STRING);
+            if (var.getScale().getDataType().equals(BrAPITraitDataType.ORDINAL)) {
+                obsVarColumn.setDataType(Column.ColumnDataType.INTEGER);
```

Comment:
I think this has to be a string for ordinal.  At least on the DeltaBreed UI, we don't enforce that the value of the ordinal be numeric.
<img width="234" alt="Screenshot 2023-06-06 at 3 26 09 PM" src="https://github.com/Breeding-Insight/bi-api/assets/3004635/bdd3705e-fe50-4d18-820f-6fe7d269bb92">

Reply:
done

Reply:
done